packages/authentication/src/__tests__/acceptance/jwt-auth-extension.acceptance.ts

```diff
@@ -0,0 +1,406 @@
+        const joeUser = this.users.list['joe@example.com'].user;
+
+        const joeUserProfile = {
+          id: joeUser.id,
+          email: joeUser.email,
+          name: `${joeUser.firstname} ${joeUser.surname}`,
+        };
+
+        return await this.tokenService.generateToken(joeUserProfile);
+      }
+
+      @get('/whoAmI')
+      @authenticate('jwt')
+      whoAmI(
+        @inject(AuthenticationBindings.CURRENT_USER) userProfile: UserProfile,
+      ) {
+        if (userProfile) {
+          if (userProfile.email) return userProfile.email;
+          else return 'userProfile email is undefined';
+        } //if
+        else return 'userProfile is undefined';
+      }
+    }
+
+    app.controller(InfoController);
+
+    const the_token: string = (await whenIMakeRequestTo(server)
+      .get('/login')
+      .expect(200)).text;
+
+    expect(the_token !== null).to.equal(true);
+    expect(typeof the_token === 'string').to.equal(true);
+
+    const email = (await whenIMakeRequestTo(server)
+      .get('/whoAmI')
+      .set('Authorization', 'Bearer ' + the_token)
+      .expect(200)).text;
+
+    expect(email).to.equal(test_users.list['joe@example.com'].user.email);
+  });
+
+  it(`returns error for missing Authorization header`, async () => {
+    class InfoController {
+      constructor(
+        @inject(JWTAuthenticationStrategyBindings.TOKEN_SERVICE)
+        public tokenService: JWTService,
+        @inject(USER_REPO)
+        public users: UserRepository,
+      ) {}
+
+      @get('/login')
+      async logIn() {
+        //
+        // ...Other code for verifying a valid user (e.g. basic or local strategy)...
+        //
+
+        // Now with a valid userProfile, let's create a JSON web token
+        const joeUser = this.users.list['joe@example.com'].user;
+
+        const joeUserProfile = {
+          id: joeUser.id,
+          email: joeUser.email,
+          name: `${joeUser.firstname} ${joeUser.surname}`,
+        };
+
+        return await this.tokenService.generateToken(joeUserProfile);
+      }
+
+      @get('/whoAmI')
+      @authenticate('jwt')
+      whoAmI(
+        @inject(AuthenticationBindings.CURRENT_USER) userProfile: UserProfile,
+      ) {
+        if (userProfile) {
+          if (userProfile.email) return userProfile.email;
+          else return 'userProfile email is undefined';
+        } //if
+        else return 'userProfile is undefined';
+      }
+    }
+
+    app.controller(InfoController);
+
+    const the_token: string = (await whenIMakeRequestTo(server)
+      .get('/login')
+      .expect(200)).text;
+
+    expect(the_token !== null).to.equal(true);
+    expect(typeof the_token === 'string').to.equal(true);
+
+    //not passing in 'Authorization' header
+    await whenIMakeRequestTo(server)
+      .get('/whoAmI')
+      .expect(401);
+  });
+
+  it(`returns error for invalid 'Bearer ' portion of Authorization header value`, async () => {
+    class InfoController {
```

Comment:
Consider moving `InfoController` to the bottom of the test as a helper if it can be re-used throughout these tests. Feel free to do it in a follow up PR.

Reply:
The Info controllers are different for pretty much all tests, so I will leave as is.